### PR TITLE
feat: export user data

### PR DESCRIPTION
**Pull request by nadduli**
### What does the PR do?
This PR introduces a new API endpoint to export user data in JSON format. It includes server-side logic for retrieving user data along with their related organisations and converting it into JSON format. 

#### How should this be manually tested?
Send a GET request to `/api/v1/export/user` with the query parameter `userId` set to a valid user ID. 

#### Example: 
GET /api/v1/export/user?userId=12345678-1234-1234-1234-123456789012

#### Checklist of what you did:
- [x] Developed the service logic to retrieve user data including related organisations.
- [x] Implemented the endpoint to export user data as JSON.
- [x] Added Swagger documentation for the new endpoint.
- [x] Handled various error scenarios including missing user ID and user not found.
- [x] Wrote unit tests to verify that the endpoint correctly retrieves and exports user data.
- [x] Tested different scenarios to ensure the endpoint handles edge cases effectively.

#### Reference Issue:
[FEAT] Export User Data Endpoint #138

### CODE REVIEW

File: .husky/pre-commit
Comment (Heba-WebDev) [inline .husky/pre-commit:1]: Please don't change the husky folder

Comment (nadduli) [reply .husky/pre-commit:1]: okay

File: src/modules/export-user/export-user.controller.ts
Comment (Heba-WebDev) [inline src/modules/export-user/export-user.controller.ts:1]: Please avoid using res.respond, take a look at nestjs docs to see how responses are handled in nest

**Review by Heba-WebDev — CHANGES_REQUESTED** [review]
Left you comments

**Comment by nadduli** [comment]
I will work on it in the morning

**Comment by nadduli** [comment]
worked on them.

**Review by nadduli — COMMENTED** [review]
all merge conflicts resolved

**Review by davidolanrewaju — DISMISSED** [review]

**Pull request closed without merging**